tests/conftest.py

```diff
@@ -83,6 +83,25 @@ def compile_and_run(device, reset_torch_dynamo, request):
 
     if "torch_ttnn" in record:
         model, inputs, outputs = record["torch_ttnn"]
+
+        try:
```

Comment:
I am not clear why you need this try/catch block, it does not call any functions and I can’t see how it can throw.

Reply:
Is this for the case when inputs or model do not contain a key?

Reply:
Yeah, at least `requires_grad` is an attribute of `torch.Tensor` [type](https://pytorch.org/docs/stable/generated/torch.Tensor.requires_grad_.html). I've seen other models have other types for their inputs (see below), so I think this PR only supports `Tensor` input types at the moment. 

https://github.com/tenstorrent/pytorch2.0_ttnn/blob/6f78bd3a6497a8bb5246e3d46fce431cadc50c37/tests/conftest.py#L172-L178

Reply:
The try/catch is to handle the fact that `inputs.requires_grad` is invalid if `inputs` is an instance of type `collections.Mapping` or `collections.Sequence`.

In other words, we assume that the models under train test must have only one input.

Will relax the assumption if necessary.